sdk/java/src/main/java/sawtooth/sdk/processor/TransactionProcessor.java

```diff
@@ -205,6 +206,19 @@ public void run() {
       if (!this.handlers.isEmpty()) {
         this.currentMessage = this.stream.receive();
         if (this.currentMessage != null) {
+          if (this.currentMessage.getMessageType() == Message.MessageType.TP_PING) {
```

Comment:
Should there be a similar check after this to only handle `TP_PROCESS_REQUEST` messages?